planning/behavior_path_planner/src/behavior_path_planner_node.cpp

```diff
@@ -54,9 +54,15 @@ BehaviorPathPlannerNode::BehaviorPathPlannerNode(const rclcpp::NodeOptions & nod
   using std::chrono_literals::operator""ms;
 
   // data_manager
+  lane_change_param_ptr = std::make_shared<LaneChangeParameters>(getLaneChangeParam());
+  avoidance_param_ptr = std::make_shared<AvoidanceParameters>(getAvoidanceParam());
   {
+    auto common_param = getCommonParam();
+    common_param.minimum_lane_change_velocity = lane_change_param_ptr->minimum_lane_change_velocity;
```

Comment:
@zulfaqar-azmi-t4 Please remove `minimum_lane_change` related params from lane_change parameter if we have the param in `common_param`.
Another option is, please move parameter handling code like this into `getXXXParam()` function. The `minimum_lane_change_*` parameter is in the lane change namespace, and the common parameter can get the param from the lane change namespace. My image is like below.

In `getCommonParam()`
```
...
common_param.minimum_lane_change_prepare_distance =
  declare_parameter("lane_change/minimum_lane_change_prepare_distance", 2.0);
```

and in `getLaneChangeParam()`
```
...
lane_change_param.minimum_lane_change_prepare_distance =
  get_parameter("minimum_lane_change_prepare_distance");  // just get since it has already been declared in common param.
```

Reply:
The aim is to keep this simple style for parameter handling:

```
planner_data_->parameters = getCommonParam();

auto side_shift_module =
  std::make_shared<SideShiftModule>("SideShift", *this, getSideShiftParam());

auto avoidance_module =
  std::make_shared<AvoidanceModule>("Avoidance", *this, getAvoidanceParam());

auto lane_following_module =
  std::make_shared<LaneFollowingModule>("LaneFollowing", *this, getLaneFollowingParam());
```

Reply:
fixed in this [6d04c4f](https://github.com/autowarefoundation/autoware.universe/pull/2428/commits/6d04c4f1c46a58f0fa02d17e88a14b6ddfb0130b)

The approach i take is calling the parameter

```
common_param.minimum_lane_change_prepare_distance =
  declare_parameter("lane_change/minimum_lane_change_prepare_distance", 2.0);
```

so that every lane change related can be configured in lane change param yaml.

Reply:
got it. Thank you!!